pxr/usd/pcp/node.h

```diff
@@ -372,8 +369,28 @@ class PcpNodeRef_ChildrenIterator
     PCP_API
     PcpNodeRef_ChildrenIterator(const PcpNodeRef& node, bool end = false);
 
```

Comment:
I'm pretty confident that you will need operator-> for both of these iterator classes. They're exposed publicly as child_const_iterator and child_const_reverse_iterator (via typedef) and I can confirm that at least the maya open source code uses these iterators and dereferences via ->